mollie/refunds.go

```diff
@@ -0,0 +1,243 @@
+package mollie
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/go-querystring/query"
+)
+
+// Refund describe a refund for a certain payment.
+type Refund struct {
+	Resource         string        `json:"resource,omitempty"`
+	ID               string        `json:"id,omitempty"`
+	Amount           *Amount       `json:"amount,omitempty"`
+	SettlementID     string        `json:"settlementId,omitempty"`
+	SettlementAmount *Amount       `json:"settlementAmount,omitempty"`
+	Description      string        `json:"description,omitempty"`
+	Metadata         interface{}   `json:"metadata,omitempty"`
+	Status           *RefundStatus `json:"status,omitempty"`
+	Lines            *OrderLines   `json:"lines,omitempty"` //TODO: this property should be from order.go
+	PaymentID        string        `json:"paymentId,omitempty"`
+	OrderID          string        `json:"orderId,omitempty"`
+	CreatedAt        *time.Time    `json:"createdAt,omitempty"`
+	Links            *RefundLinks  `json:"_links,omitempty"`
+}
+
+// RefundList describes how a list of refunds will be retrieved by Mollie.
+type RefundList struct {
+	Count    int `json:"count,omitempty"`
+	Embedded struct {
+		Refunds []Refund
+	} `json:"_embedded,omitempty"`
+	Links PaginationLinks `json:"_links,omitempty"`
+}
+
+// RefundStatus describes the status of the refund.
+type RefundStatus string
+
+// Valid refund status.
+const (
+	Queued     RefundStatus = "queued"
+	Pending    RefundStatus = "pending"
+	Processing RefundStatus = "processing"
+	Refunded   RefundStatus = "refunded"
+	Failed     RefundStatus = "failed"
+)
+
+// OrderLines describes an array of order line objects.
+type OrderLines struct {
+	Quantity       int     `json:"quantity,omitempty"`
+	DiscountAmount *Amount `json:"discountAmount,omitempty"`
+	VatAmount      *Amount `json:"vatAmount,omitempty"`
+	TotalAmount    *Amount `json:"totalAmount,omitempty"`
+}
+
+// RefundLinks describes all the possible links to be returned with
+// a Refund object.
+type RefundLinks struct {
+	Self       *URL `json:"self,omitempty"`
+	Payment    *URL `json:"payment,omitempty"`
+	Settlement *URL `json:"settlement,omitempty"`
+	Order      *URL `json:"order,omitempty"`
+}
+
+// RefundOptions describes refund endpoint valid query string parameters.
+//
+// See: https://docs.mollie.com/reference/v2/refunds-api/get-refund.
+type RefundOptions struct {
+	Embed    EmbedValue `url:"embed,omitempty"`
+	TestMode bool       `url:"testmode,omitempty"`
```

Comment:
Hey @dwikur21 the test mode param is added as a first-class para when creating a new requests, the relevant code is here: https://github.com/VictorAvelar/mollie-api-go/blob/e13fd309c27adf67ccc177a3c665855be17e8c47/mollie/mollie.go#L80

Reply:
Didn't see it. Ok, I'll delete it.